multicluster/charts/linkerd-multicluster/templates/gateway.yaml

```diff
@@ -29,10 +34,32 @@ spec:
       labels:
         app: {{.Values.gateway.name}}
     spec:
+      {{- if .Values.enablePodAntiAffinity -}}
+      {{- $local := dict "component" "gateway" -}}
+      {{- include "linkerd.pod-affinity" $local | nindent 6 -}}
```

Comment:
I believe this partial uses the `linkerd.io/control-plane-component` label to control anti-affinity.  Unfortunately, that won't work here because we don't use that label because this isn't part of the Linkerd control plane.  Probably we'd need to use a different label.

Reply:
As per @adleong 's comments above I've added the `linkerd.io/control-plane-component: gateway` label to the pods which should resolve this, right?

Reply:
@Crevil Actually I agree with @adleong , it might not be a good idea to use `linkerd.io/control-plane-component` and leave that to the core (control-plane) components only. I don't see why not use `app` :
```yaml
      {{- if .Values.enablePodAntiAffinity -}}
      {{- $local := dict "label" "app" "component" .Values.gateway.name -}}
      {{- include "linkerd.pod-affinity" $local | nindent 6 -}}
```

Reply:
Let's go with the app label then 👍 I'll update the PR

Reply:
I have updated the PR now. We now use the existing `app` label to select pod affinity. Further more I have removed the introduced `control-plane-component` label again 👍